src/python/pants/option/global_options.py

```diff
@@ -1989,7 +1989,12 @@ def resolve_keep_sandboxes(
 
         if isinstance(resolved_value, bool):
             # Is `process_cleanup`.
-            return KeepSandboxes.always if resolved_value else KeepSandboxes.on_failure
+            warn_or_error(
+                removal_version="2.15.0.dev1",
+                entity="--process-cleanup",
+                hint="Instead, use `--keep-sandboxes`.",
+            )
```

Comment:
The option itself is deprecated with a start version: https://github.com/pantsbuild/pants/blob/c431a00cb9456db3a0699be68d9ba9cd50270103/src/python/pants/option/global_options.py#L1160-L1170

...so it was not supposed to be rendering yet. Using the start version was somewhat arbitrary, but was in part due to there having been a lot of discussion recently around avoiding aggressively deprecating widely-used things.

Reply:
The lack of any warning threw me for a loop, because worked yesterday did not work today with no warning.

Reply:
With the fix to the value-when-explicitly-specified, how would you feel about this going back to relying on the `deprecation_start_version` instead?

Reply:
I have no feeling about that. I was just getting broken behavior and it took me a while to figure out. As long as no one else is put in that position I'm happy.